src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js

```diff
@@ -0,0 +1,352 @@
+
+                if (hostIndex === -1) {
+                    vm.selectedHost.push(host);
+                } else {
+                    vm.selectedHost.splice(hostIndex, 1);
+                }
+            }
+
+            _calculateSummaryValues();
+            vm.getMaxCount();
+        }
+
+        /** Checks whether host is selected or not **/
+        function isSelectedHost(host) {
+            return vm.selectedHost.indexOf(host) > -1;
+        }
+
+        /** Filer the host list based on the parameters **/
+        function filterList(item) {
+            var properties,
+                property,
+                i,
+                diskLen,
+                disks,
+                searchBy = {},
+                eth,
+                ethlen,
+                list = item.ifIPMapping,
+                len = list.length;
+
+            if (vm.hostFilterBy.value) {
+
+                properties = vm.hostFilterBy.property.split(".");
+
+                if (properties.length > 1) {
+                    property = properties[1];
+
+                    for (i = 0; i < len; i++) {
+                        if (list[i][property].toLowerCase().indexOf(vm.hostFilterBy.value.toLowerCase()) >= 0) {
+                            item.isExpanded = true;
+                            return item;
+                        }
+                    }
+
+                } else {
+                    property = vm.hostFilterBy.property;
+
+                    if (item[property].toLowerCase().indexOf(vm.hostFilterBy.value.toLowerCase()) >= 0) {
+                        return item;
+                    }
+                }
+
+            } else {
+                return item;
+            }
+        }
+
+        /* Private Functions */
+
+        /** To check whether the nodes are in the same public network in which cluster was created**/
+        function _isInSelectedNetwork(host) {
+            var interfaceKeys = Object.keys(host.networks),
+                len = interfaceKeys.length,
+                flag = false,
+                i,
+                ip,
+                ipLen,
+                j;
+
+            for (i = 0; i < len; i++) {
+
+                if(host.networks[interfaceKeys[i]].subnet === selectedCluster.clusterNetwork) {
+                    flag = true;
+                    host.provisioningIP = host.networks[interfaceKeys[i]].ipv4[0];
+                    return flag;
+                }
+            }
+
+            return flag;
+        }
+
+        /** Change the host.networks to JSON **/
+        function _changeIntoArray(host) {
+            var i,
+                interfaces = Object.keys(host.networks),
+                len = interfaces.length;
+
+            for (i = 0; i < len; i++) {
+                host.networks[interfaces[i]].ipv4 = JSON.parse(host.networks[interfaces[i]].ipv4);
+            }
+        }
+
+        /** Create host list after manipulating the API response **/
+        function _createHostList(host) {
```

Comment:
To large function, needs to be divided into smaller functions.

https://softwareengineering.stackexchange.com/questions/133404/what-is-the-ideal-length-of-a-method-for-you